File: tien-len-server/game/deck.js
const Card = require('./card'); // Import Card from card.js

class Deck {
    constructor() {
        this.suits = ["spades", "clubs", "diamonds", "hearts"];
        this.ranks = ["3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King", "Ace", "2"];
        this.cards = [];
    }

    initializeDeck() {
        for (const suit of this.suits) {
            for (const rank of this.ranks) {
                this.cards.push(new Card(suit, rank));
            }
        }
    }

    shuffleDeck() {
        for (let i = this.cards.length - 1; i > 0; i--) {
            const j = Math.floor(Math.random() * (i + 1));
            [this.cards[i], this.cards[j]] = [this.cards[j], this.cards[i]];
        }
    }

    dealToPlayers(numPlayers, numCardsPerPlayer) {
        const hands = [];
        for (let i = 0; i < numCardsPerPlayer; i++) {
            for (let j = 0; j < numPlayers; j++) {
                const card = this.cards.pop();
                hands[j].push(card);
            }
        }

        return hands;
    }
}

module.exports = Deck;

File: java-cli/server/src/main/java/Deck.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Deck {
	private static Deck instance;

	public static final String[] SUITS_ORDER = { "spades", "clubs", "diamonds", "hearts" };
	public static final String[] RANKS_ORDER = { "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King",
			"Ace","2" };
	public static final String[] CARDS_ORDER = { "spades3", "clubs3", "diamonds3", "hearts3", "spades4", "clubs4",
			"diamonds4", "hearts4", "spades5", "clubs5", "diamonds5", "hearts5", "spades6", "clubs6", "diamonds6",
			"hearts6", "spades7", "clubs7", "diamonds7", "hearts7", "spades8", "clubs8", "diamonds8", "hearts8",
			"spades9", "clubs9", "diamonds9", "hearts9", "spades10", "clubs10", "diamonds10", "hearts10", "spadesJack",
			"clubsJack", "diamondsJack", "heartsJack", "spadesQueen", "clubsQueen", "diamondsQueen", "heartsQueen",
			"spadesKing", "clubsKing", "diamondsKing", "heartsKing", "spadesAce", "clubsAce", "diamondsAce",
			"heartsAce", "spades2", "clubs2", "diamonds2", "hearts2" };

	private List<Card> cards;

	private Deck() {
		super();
		this.cards = new ArrayList<Card>();
		initDeck();
	}

	void initDeck() {
		for (int i = 0; i < SUITS_ORDER.length; i++) {
			for (int j = 0; j < RANKS_ORDER.length; j++) {
//				System.out.println(SUITS_ORDER[i] + RANKS_ORDER[j]);
				this.cards.add(new Card(SUITS_ORDER[i], RANKS_ORDER[j]));
			}
		}
	}

	public static Deck getInstance() {
		if (instance == null) {
			instance = new Deck();
		}
		return instance;
	}

	public void shufle() {
		Collections.shuffle(cards);
	}

	public List<Card> getCards() {
		return cards;
	}
	
}
